src/main/java/com/artipie/asto/memory/InMemoryStorage.java

```diff
@@ -46,16 +47,36 @@
     /**
      * Values stored by key strings.
      */
-    private final Map<String, byte[]> data = new HashMap<>();
+    private final NavigableMap<String, byte[]> data;
+
+    /**
+     * Ctor.
+     */
+    public InMemoryStorage() {
+        this.data = new TreeMap<>();
```

Comment:
@fanifieiev good point about thread safety! We should really make this class thread safe. Though making Map synchronized won't be enough, because there multiple operations with the map in many methods. So I added sync blocks to every methods (even not implemented ones as a reminder)